highfidelity/hifi
# Moved audio asset processing to a separate thread

Should fix https://highfidelity.fogbugz.com/f/cases/4857/Move-sounds-on-to-a-different-thread-so-they-don-t-lock-up-rendering

Test Plan

1. Start interface
2. Visit an area or do something where you will be forced to download new audio files
3. Verify that the audio files have been obtained and that they play correctly
4. Compare loading times of this PR vs. master, verify that this PR finishes loads scene faster when there are a significant amount of sound assets that need to be fetched (see @ZappoMan's comment below)
